### ISSUE TITLE
script-runner@2.2.2 failed to install

### ISSUE THREAD

**Issue by OMALX** [ioquatix/script-runner#87]
I will admit, I am a noob. So there's a high possibility I am doing something wrong. However, I simply can't figure it out. I have had script-runner previously installed and it worked beautifully after the latest up though I haven't been able to get it to work. I uninstalled it hoping to re-install it and seeing if that would fix my issue. It did not and now I can't even get it to install. Am I missing something or is it something else? 

Installing “script-runner@2.2.2” failed.Hide output…

> node-pty@0.7.4 install /private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr/node_modules/script-runner/node_modules/node-pty
> node scripts/install.js

/private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr
`-- (empty)

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 17.4.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr/node_modules/script-runner/node_modules/node-pty
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr/package.json'
npm WARN apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr No description
npm WARN apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr No repository field.
npm WARN apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr No README data
npm WARN apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr No license field.
npm ERR! Darwin 17.4.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/m/.atom/.apm/.apmrc" "--userconfig" "/Users/m/.atom/.apmrc" "install" "/private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/d-11815-2024-1uvloic.s1fh41jor/package.tgz" "--runtime=electron" "--target=1.6.15" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! node-pty@0.7.4 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-pty@0.7.4 install script 'node scripts/install.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-pty package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/install.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-pty
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-pty
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/l0/4kxf1wks09j_zwgrbhq7y7q80000gn/T/apm-install-dir-11815-2024-1luldzp.zhhlqpk3xr/npm-debug.log
npm ERR! code 1

**Comment by ioquatix** [comment]
This looks like an issue with Xcode.

What does: `Xcode-select --install` do?

**Comment by SolusPrime** [comment]
> node-pty@0.7.4 install /tmp/apm-install-dir-11819-5971-1d4lk5/node_modules/script-runner/node_modules/node-pty
> node scripts/install.js


npm WARN engine temp-write@3.4.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine make-dir@1.1.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine pify@3.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine temp-dir@1.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine os-locale@2.1.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine execa@0.7.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine mem@1.1.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine mimic-fn@1.2.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine p-finally@1.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine get-stream@3.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine npm-run-path@2.0.2: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})

/tmp/apm-install-dir-11819-5971-1d4lk5/node_modules/script-runner/node_modules/node-pty/scripts/install.js:3
const os = require('os');
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3
npm WARN engine path-key@2.0.1: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm ERR! Linux 3.16.0-45-generic
npm ERR! argv "/opt/atom/resources/app/apm/bin/node" "/opt/atom/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/home/bodhi/.atom/.apm/.apmrc" "--userconfig" "/home/bodhi/.atom/.apmrc" "install" "/tmp/d-11819-5971-16zb9wc/package.tgz" "--target=0.22.0" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! node-pty@0.7.4 install: `node scripts/install.js`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the node-pty@0.7.4 install script 'node scripts/install.js'.
npm ERR! This is most likely a problem with the node-pty package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls node-pty
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /tmp/apm-install-dir-11819-5971-1d4lk5/npm-debug.log

**Comment by ioquatix** [comment]
These are two completely different problems.

@SolusPrime I think your node version is out of date. But if anything, it's not a problem with `script-runner` but a problem with `node-pty`.

**Comment by OMALX** [comment]
@ioquatix 

I tried to fix the Xcode issue, but I am still getting the same results.

**Comment by ioquatix** [comment]
What issue did you try to fix and how did you do it?

**Comment by OMALX** [comment]
You said it looks like an issue with Xcode then asked "What does: xcode-select --install do?"

I tried that, got the same result.

**Comment by bjce** [comment]
I think I'm another noob but I also get the same kind of message while trying to install script-runner. Here is the output: 

```
Installing “script-runner@2.3.0” failed.Hide output…

> node-pty@0.7.4 install /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/node_modules/script-runner/node_modules/node-pty
> node scripts/install.js

/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i
`-- (empty)


gyp ERR! configure error 
gyp ERR! stack Error: Python executable "/Applications/Atom.app/Contents/Resources/app/apm/bin/python-interceptor.sh" is v3.6.4, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at failPythonVersion (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:454:14)
gyp ERR! stack     at /Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:443:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:197:7)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:334:11)
gyp ERR! stack     at emitOne (events.js:96:13)
gyp ERR! stack     at Socket.emit (events.js:188:7)
gyp ERR! stack     at Pipe._handle.close [as _onclose] (net.js:498:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/node_modules/script-runner/node_modules/node-pty
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/package.json'
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No description
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No repository field.
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No README data
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No license field.
npm ERR! Darwin 17.5.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/mymac/.atom/.apm/.apmrc" "--userconfig" "/Users/mymac/.atom/.apmrc" "install" "/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/d-118410-56277-1qh3olt.zpu2qpvi/package.tgz" "--runtime=electron" "--target=1.7.11" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! node-pty@0.7.4 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-pty@0.7.4 install script 'node scripts/install.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-pty package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/install.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-pty
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-pty
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/npm-debug.log
npm ERR! code 1
```

**Comment by Gkamble** [comment]
I also have the same problem as above, 


Installing “script-runner@2.3.0” failed.Hide output…

> node-pty@0.7.4 install /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/node_modules/script-runner/node_modules/node-pty
> node scripts/install.js

/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i
`-- (empty)


gyp ERR! configure error 
gyp ERR! stack Error: Python executable "/Applications/Atom.app/Contents/Resources/app/apm/bin/python-interceptor.sh" is v3.6.4, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at failPythonVersion (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:454:14)
gyp ERR! stack     at /Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:443:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:197:7)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:334:11)
gyp ERR! stack     at emitOne (events.js:96:13)
gyp ERR! stack     at Socket.emit (events.js:188:7)
gyp ERR! stack     at Pipe._handle.close [as _onclose] (net.js:498:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/node_modules/script-runner/node_modules/node-pty
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/package.json'
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No description
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No repository field.
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No README data
npm WARN apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i No license field.
npm ERR! Darwin 17.5.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/mymac/.atom/.apm/.apmrc" "--userconfig" "/Users/mymac/.atom/.apmrc" "install" "/private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/d-118410-56277-1qh3olt.zpu2qpvi/package.tgz" "--runtime=electron" "--target=1.7.11" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! node-pty@0.7.4 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-pty@0.7.4 install script 'node scripts/install.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-pty package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/install.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-pty
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-pty
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/jq/yyp9q2fs1z5780k9l1_2ph4r0000gn/T/apm-install-dir-118410-56277-1rfdvd2.srtfomkj4i/npm-debug.log
npm ERR! code 1



HELP!

**Comment by MarvinKweyu** [comment]
Same problem Installing “script-runner@2.3.0” failed." Hoped for a solution here.Is there no possible solution available! I've been updating my distro in hopes that that might help solve it....null

**Comment by huzaifabinsaif** [comment]
same problem installing script runner

**Comment by huba** [comment]
Righto, a fix is in the works coming to a release v2.3.1 near you.

**Comment by huzaifabinsaif** [comment]
Thank you for looking into it.
